### PR TITLE
git-sync: add page

### PR DESCRIPTION
**Pull request by deadpyxel**
- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

**Version of the command being documented (if known):** 6.2.0

Opened from #5137

### CODE REVIEW

File: pages/common/git-sync.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# git sync
+
+> Sync local branch with remote branch.
```
Comment (navarroaxel) [inline pages/common/git-sync.md:3]:
```suggestion
> Sync local branches with remote branches.
```

File: pages/common/git-sync.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+> Part of `git-extras`.
+> More information: <https://github.com/tj/git-extras/blob/master/Commands.md#git-sync>.
+
+- Sync local branch with its remote branch:
```
Comment (navarroaxel) [inline pages/common/git-sync.md:7]:
```suggestion
- Sync the current local branch with its remote branch:
```

File: pages/common/git-sync.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+`git sync`
+
+- Sync local branch with origin/master:
```
Comment (navarroaxel) [inline pages/common/git-sync.md:11]: I think we should use `main` for this new Git era
```suggestion
- Sync the current local branch with the remote main branch:
```

File: pages/common/git-sync.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+`git sync origin master`
```
Comment (navarroaxel) [inline pages/common/git-sync.md:13]:
```suggestion
`git sync origin main`
```

File: pages/common/git-sync.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+- Sync without cleaning untracked files:
+
+`git sync -s origin master`
```
Comment (navarroaxel) [inline pages/common/git-sync.md:17]: I think we should generalize this example
```suggestion
`git sync -s {{remote_name}} {{branch_name}}`
```

**Review by navarroaxel — CHANGES_REQUESTED** [review]

File: pages/common/git-sync.md (hunk 4 above)
Comment (deadpyxel) [reply pages/common/git-sync.md:13]: I keep forgetting that the standard name for default branches has been `main` for some time.

**Comment by deadpyxel** [comment]
Thanks for the suggestions, I have applied then and reviewed some examples from the documentation and manpage.

**Review by navarroaxel — APPROVED** [review]

**Review by CleanMachine1 — APPROVED** [review]

**Pull request merged**